docs/less-coding-guidelines.md

```diff
@@ -0,0 +1,393 @@
+  * [font-weight](#fontweight)
+  * [line-height](#lineheight)
+  * [letter-spacing](#letterspacing)
+* [Polyfills](#polyfills)
+* [Formatting](#formatting)
+  * [Spacing](#spacing)
+  * [Quotes](#quotes)
+* [Performance](#performance)
+  * [Specificity](#specificity)
+
+
+<a name="javascript"></a>
+## JavaScript
+
+syntax: `js-<targetName>`
+
+JavaScript-specific classes reduce the risk that changing the structure or theme of components will inadvertently affect any required JavaScript behaviour and complex functionality. It is not neccesarry to use them in every case, just think of them as a tool in your utility belt. If you are creating a class, which you dont intend to use for styling, but instead only as a selector in JavaScript, you should probably be adding the `js-` prefix. In practice this looks like this:
+
+```html
+<a href="/login" class="btn btn-primary js-login"></a>
+```
+
+**Again, JavaScript-specific classes should not, under any circumstances, be styled.**
+
+<a name="utilities"></a>
+## Utilities
+
+Medium's utility classes are low-level structural and positional traits. Utilities can be applied directly to any element; multiple utilities can be used together; and utilities can be used alongside component classes.
+
+Utilities exist because certain CSS properties and patterns are used frequently. For example: floats, containing floats, vertical alignment, text truncation. Relying on utilities can help to reduce repetition and provide consistent implementations. They also act as a philosophical alternative to functional (i.e. non-polyfill) mixins.
+
+
+```html
+<div class="u-clearfix">
+  <p class="u-textTruncate">{$text}</p>
+  <img class="u-pullLeft" src="{$src}" alt="">
+  <img class="u-pullLeft" src="{$src}" alt="">
+  <img class="u-pullLeft" src="{$src}" alt="">
+</div>
+```
+
+<a name="u-utilityName"></a>
+### u-utilityName
+
+Syntax: `u-<utilityName>`
+
+Utilities must use a camel case name, prefixed with a `u` namespace. What follows is an example of how various utilities can be used to create a simple structure within a component.
+
+```html
+<div class="u-clearfix">
+  <a class="u-pullLeft" href="{$url}">
+    <img class="u-block" src="{$src}" alt="">
+  </a>
+  <p class="u-sizeFill u-textBreak">
+    â€¦
+  </p>
+</div>
+```
+
+<a name="components"></a>
+## Components
+
+Syntax: `<componentName>[--modifierName|-descendantName]`
+
+Component driven development offers several benefits when reading and writing HTML and CSS:
+
+* It helps to distinguish between the classes for the root of the component, descendant elements, and modifications.
+* It keeps the specificity of selectors low.
+* It helps to decouple presentation semantics from document semantics.
+
+You can think of components as custom elements that enclose specific semantics, styling, and behaviour.
+
+
+<a name="componentName"></a>
+### ComponentName
+
+The component's name must be written in camel case.
+
+```css
+.myComponent { /* â€¦ */ }
```

Comment:
Looks like some character encoding issues. Can you resave this file with utf8 encoding and see if this disappears?